src/connection-manager/connections.ts

```diff
@@ -26,89 +32,22 @@ const network_fee: Map<string, Fee> = new Map()
 const CM_INTERVAL = 60 * 60 * 1000
 const WS_TIMEOUT = 10000
 const REPORTING_INTERVAL = 15 * 60 * 1000
-const LEDGER_HASHES_SIZE = 10
 let cmStarted = false
 
-/**
- * Subscribes a WebSocket to manifests and validations streams.
- *
- * @param ws - A WebSocket object.
- */
-function subscribe(ws: WebSocket): void {
-  ws.send(
-    JSON.stringify({
-      id: 2,
-      command: 'subscribe',
-      streams: ['manifests', 'validations', 'ledger'],
-    }),
-  )
-}
-
-/**
- * Handles a WebSocket message received.
- *
- * @param data - The WebSocket message received from connection.
- * @param ledger_hashes - The list of recent ledger hashes.
- * @param networks - The networks of subscribed node.
- * @returns Void.
- */
-async function handleWsMessageTypes(
-  data: ValidationRaw | StreamManifest | StreamLedger,
-  ledger_hashes: string[],
-  networks: string | undefined,
-): Promise<void> {
-  if (data.type === 'validationReceived') {
-    const validationData = data as ValidationRaw
-    if (ledger_hashes.includes(validationData.ledger_hash)) {
-      validationData.networks = networks
-    }
-
-    // Get network of the validation if ledger_hash is not in cache.
-    const validationNetworkDb: DatabaseValidator | undefined = await query(
-      'validators',
-    )
-      .select('*')
-      .where('signing_key', validationData.validation_public_key)
-      .first()
-    const validationNetwork =
-      validationNetworkDb?.networks ?? validationData.networks
-
-    // Get the fee for the network to be used in case the validator does not vote for a new fee.
-    if (validationNetwork) {
-      validationData.ledger_fee = network_fee.get(validationNetwork)
-    }
-    void agreement.handleValidation(validationData)
-  } else if (data.type === 'manifestReceived') {
-    void handleManifest(data as StreamManifest)
-  } else if (data.type.includes('ledger')) {
-    const current_ledger = data as StreamLedger
-    ledger_hashes.push(current_ledger.ledger_hash)
-    if (networks) {
-      const fee: Fee = {
-        fee_base: current_ledger.fee_base,
-        reserve_base: current_ledger.reserve_base,
-        reserve_inc: current_ledger.reserve_inc,
-      }
-      network_fee.set(networks, fee)
-    }
-    if (ledger_hashes.length > LEDGER_HASHES_SIZE) {
-      ledger_hashes.shift()
-    }
-  }
-}
-
 /**
  * Sets the handlers for each WebSocket object.
  *
  * @param ip - The ip address of the node we are trying to reach.
  * @param ws - A WebSocket object.
  * @param networks - The networks of the node we are trying to reach where it retrieves validations.
+ * @param isInitialNode - Whether source node is an entry/initial node for the network.
```

Comment:
I'm not sure what an entry node is in this case - is it an XRPL specific term, or is it just the first node you've encountered when crawling? (If there's a docs page which is relevant, can you link to it?)

Reply:
From reading the code more, it seems this field is used to look up manually added values - in which case I would rename this to be more "functional" - meaning something like `lookUpAmendmentsManually`. That way it's easier to process this single function's purpose on it's own without knowing the architecture of the project (e.g. that we start by looking up nodes with amendment data we pre-defined)

Reply:
For this part I want to look for fee data to apply the fee for all validators in a network that doesn't vote for a new fee, and I want to do it only once per network. That's why I added a boolean to check if the node is the initial node that we added manually at first for each network in order to start crawling, so the the ledgerEntry calls only execute once per network per ledger. Sorry for it being confusing for you

Reply:
I see, that makes sense :) 

Can you add an in-line comment above `if (isInitialNode) {` explaining that?

My thought process: In general, I tend to think if there's a question a code reviewer has that seems reasonable, but the code is working as is the resolution is to make that same answer more discoverable in the code. 

The best way to do that is to simplify the logic. Here, since you're only using this variable once, I checked if it was reasonable to pull that parameter / action out of this function so it can be called once separately. In this case I don't think that makes sense.

Then since the logic seems fine, a comment where the confusion arose seems like a reasonable answer.